### PR TITLE
up-to-date description of the Horace installation on IDAAAS …

### PR DESCRIPTION
**Pull request by abuts**
…and the release/bugfix process.

### CODE REVIEW

File: documentation/smg/11_ISIS_release.md [hunk 1]
```diff
@@ -1,27 +1,43 @@
 # ISIS Release Process
 
-Horace is deployed separately to a number of systems across ISIS
+Horace deployed on iDAAaS system and machines dedicated for excitations as a single git repository available
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:3]:
```suggestion
Horace is deployed on iDAAaS system and machines dedicated for Excitations as a single git repository available 
```

File: documentation/smg/11_ISIS_release.md [hunk 2]
```diff
@@ -1,27 +1,43 @@
 # ISIS Release Process
 
-Horace is deployed separately to a number of systems across ISIS
+Horace deployed on iDAAaS system and machines dedicated for excitations as a single git repository available 
+read-only for excitation machine users and providing read/write access for the members of the excitations group.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:4]:
```suggestion
read-only for Excitations machine users and providing read/write access for the members of the Excitations group.
```

File: documentation/smg/11_ISIS_release.md [hunk 3]
```diff
@@ -1,27 +1,43 @@
...
 
-## ISIS Compute
+As the single place of the Horace code is available to all machines, the changes to the code made by releasers are immediately available to all iDAAaS machines.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:6]:
```suggestion
As the single place where the Horace code is available to all machines, the changes to the code made by releasers are immediately available to all iDAAaS machines.
```

File: documentation/smg/11_ISIS_release.md [hunk 4]
```diff
@@ -1,27 +1,43 @@
...
 
-#### Contact
+## Horace installation in ISIS
+
+Common path `/usr/local/mprogs/Users` is added to the search path of all MATLAB sessions available to users processing the results of inelastic neutron scattering experiments in ISIS.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:10]:
```suggestion
A common filepath `/usr/local/mprogs/Users` is added to the search path of all MATLAB sessions available to users processing the results of inelastic neutron scattering experiments in ISIS.
```

File: documentation/smg/11_ISIS_release.md [hunk 5]
```diff
@@ -1,27 +1,43 @@
...
+ 
+On the machines, managed by excitation group this is achieved by adding this path to the end of the `{MATLAB Installation Folder}/toolbox/local/pathdef.m` file, though recent iDAAaS virtual machines seems use different approach for adding this path to the MATLAB search path.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:12]:
```suggestion
On the machines managed by the Excitations group, this is achieved by adding this path to the end of the list of paths referenced in the`{MATLAB Installation Folder}/toolbox/local/pathdef.m` file. 

Note that recent iDAaaS virtual machines use a different approach for adding this path to the MATLAB search path.
```

There needs to be more detail about the different approach. We can't have an instruction that doesn't work on half the IDAaaS machines.

File: documentation/smg/11_ISIS_release.md [hunk 6]
```diff
@@ -1,27 +1,43 @@
...
+
+This path contains Horace initialization script `horace_on.m` which, on execution, adds all necessary folders to MATLAB search path and initialize all other Horace variables necessary for Horace operations. (i.e. Horace configurations). This folder also contains the worker script, which invokes `horace_on` script and initializes Horace for all parallel MATLAB sessions.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:14]:
```suggestion
This path includes the Horace initialization script `horace_on.m` which, on execution, adds all necessary folders to MATLAB search path and initializes all other Horace variables necessary for Horace operations. (i.e. Horace configurations). This folder also contains the worker script which invokes the `horace_on` script and initializes Horace for all parallel MATLAB sessions. 
```
Did you mean i.e. so only configurations are initialized or did you mean e.g. so other things are initialized as well?

File: documentation/smg/11_ISIS_release.md [hunk 7]
```diff
@@ -1,27 +1,43 @@
...
-FBU IT support team: FBUitservicedesk@stfc.ac.uk 
+Administrative script (Horace developers do not have write access to it) `/var/daaas/dynamic/matlabsetup.m` is executed on start up of MATLAB user sessions of iDAAaS machines and invokes `horace_on` for user sessions dedicated for using Horace. 
 
 
-#### Deploy Location
-The physical location of the code on ISISCOMPUTE file system is:
+## Deploy Location
+The repositories with the Horace code on iDAAaS file system are currently located under:
+`/mnt/ceph/auxiliary/excitations/isis_direct_soft/` 
+Number of the actual repositories are cloned under this location:
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:22]:
```suggestion
A number of the actual repositories are cloned under this location:
```

File: documentation/smg/11_ISIS_release.md [hunk 8]
```diff
@@ -1,27 +1,43 @@
...
-`/home/isis_direct_soft/Herbert_bugfix`
+ -- `Horace_git` -- the repository prepared for recent Horace release. As we have not actually released Horace, `master` is currently checked out there. Horace-4 master have also `Herbert` code included.
+ 
+ -- `Horace_3_6` -- the repository for last release of Horace-3. `horace_3_6_3` branch containing latest bugfix release of   Horace.
+ 
+ -- `Horace_3_2` -- historical Horace release hopefully not used any more
+ 
+ -- `Horace_bugfix` -- the special repository clone, with purpose, described later.
+ 
+ As previous to version 4 versions of Horace needed correspondent versions of Herbert, three clones of Herbert repositories, corresponding to appropriate Horace repositories are also cloned here, namely: `Herbert_git`, `Herbert_3_6`, `Herbert_3_2`, and `Herbert_bugfix`
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:33]:
```suggestion
 As, previous to version 4, versions of Horace needed correspondent versions of Herbert, three clones of Herbert repositories, corresponding to appropriate Horace repositories are also cloned here, namely: `Herbert_git`, `Herbert_3_6`, `Herbert_3_2`, and `Herbert_bugfix`
```

File: documentation/smg/11_ISIS_release.md [hunk 9]
```diff
@@ -32,48 +48,44 @@ Code base for bugfix purposes is linked directly to the git repository for suppo
 `/usr/local/mprogs/Herbert_bugfix` -> `/home/isis_direct_soft/Herbert_bugfix`
 `/usr/local/mprogs/Horace_bugfix` -> `/home/isis_direct_soft/Horace_bugfix`
 
-#### Deploy Process
+Recent Horace version does not need separate Horace, so only second link should be used for running bugfix code.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:51]:
```suggestion
Recent Horace version does not need separate Herbert, so only second link should be used for running bugfix code.
```

File: documentation/smg/11_ISIS_release.md [hunk 10]
```diff
@@ -32,48 +48,44 @@ Code base for bugfix purposes is linked directly to the git repository for suppo
...
-The extraction of the release artefacts will be automated in a future, as [iDaaaS testing pipeline](https://github.com/pace-neutrons/Horace/issues/271) is implemented. The releaser then manually checks out the release branch into `Herbert_git/Horace_git` areas above and copies the release artefacts into appropriate places of the code tree.
 
-#### Notes
+## Deploy Process
 
-The script which automates these operations and performs internal releases as one-click operation will be developed in the nearest future.
+As soon as release branch is tagged and created in the repository, manual build/release process is preformed in the private area of iDaaaS machine to obtain release artefacts for the iDAAaS computational service. It is also possible to pull the release artefacts from Jenkins builds, though this process needs thorough testing. 
+The extraction of the release artefacts will be automated in a future, as [iDaaaS testing pipeline](https://github.com/pace-neutrons/Horace/issues/271) is implemented.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:58]:
```suggestion
The extraction of the release artefacts will be automated in future, as [iDaaaS testing pipeline](https://github.com/pace-neutrons/Horace/issues/271) is implemented.
```

File: documentation/smg/11_ISIS_release.md [hunk 11]
```diff
@@ -32,48 +48,44 @@ Code base for bugfix purposes is linked directly to the git repository for suppo
...
 
-The members of **mslice** group on ISISCOMPUTE have write access to release code areas. Alex Buts or FBU IT support team can currently add users to **mslice** group on ISISCOMPUTE. Normal users have read-only access to the code area, sufficient to use the code in their computations.
+The releaser then manually checks out the release branch into `Horace_git` areas above and copies the release artefacts into appropriate places of the code tree.
 
-## IDAaaS
+Minor changes and bugfixes do not need the rebuild of the release artefacts and performed by pulling recent merges to the release branch into `Horace_git` repository.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:62]: you don't say WHAT is performed. The verb needs a subject.

File: documentation/smg/11_ISIS_release.md [hunk 12]
```diff
@@ -32,48 +48,44 @@ Code base for bugfix purposes is linked directly to the git repository for suppo
...
 
-#### Contact
+## Horace-4 Transitional period
 
-iDaaaS support team : support@analysis.stfc.ac.uk
+While Horace-4 is undergoing pre-release process, it is expected to be used alongside with previous version of Horace. This is why Horace master branch is currently checked out at `Horace_git` path (recent Horace is combined with Herbert) and latest (3.6.3) version of Horace/Herbert is checked out at `Horace_3_6` and `Herbert_git` path-es.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:66]:
```suggestion
While Horace-4 is undergoing pre-release process, it is expected to be used alongside previous version of Horace. This is why Horace master branch is currently checked out at `Horace_git` path (recent Horace is combined with Herbert) and latest (3.6.3) version of Horace/Herbert is checked out at `Horace_3_6` and `Herbert_git` paths. 
```

File: documentation/smg/11_ISIS_release.md [hunk 13]
```diff
@@ -32,48 +48,44 @@ Code base for bugfix purposes is linked directly to the git repository for suppo
...
 
-#### Deploy Location
 
-physical location is the same as on ISISCOMPUTE, except ISISCOMPUTE file system is currently mounted on iDaaaS at `/mnt/nomachine`. This can change in a future, but by agreement with iDaaaS team, the symbolic links in `/usr/local/mprogs` will always point to a physical location of appropriate Horace/Herbert repository clones. 
+## Notes
 
-To ensure compatibility and smooth user's experience, all code operations should be performed with the simulink path-es, which will remain constant regardless of physical location of the code.
+ The the group which have write access to the Horace repositories is named `standarduser`, and IDAAaS support is responsible for adding users to this group. Contact supportanalysis@stfc.ac.uk to be included in this group and obtain write access to the repositories. Normal users have read-only access to the code area, sufficient to use the code in their computations.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:72]:
```suggestion
 The linux file-group which has write access to the Horace repositories is named `standarduser`, and IDAaaS support is responsible for adding users to this group. Contact supportanalysis@stfc.ac.uk to be included in this group and obtain write access to the repositories. Normal users have read-only access to the code area, sufficient to use the code in their computations.
```

File: documentation/smg/11_ISIS_release.md [hunk 14]
```diff
@@ -32,48 +48,44 @@ Code base for bugfix purposes is linked directly to the git repository for suppo
...
-
-The write access to code repository on iDaaaS is granted by iDaaaS team on request. Currently the list of people, who has the write access coincides with the members of **mslice** group on ISISCOMPUTE. This may change in a future. 
+#### Contact
 
+iDaaaS support team : support@analysis.stfc.ac.uk
 
 
 ## SCARF
 
+ Horace is currently not available on SCARF and this chapter is not verified and probably incorrect. This was working in the past and will probably be organized in some similar way in a future.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:82]:
```suggestion
 Horace is currently not available on SCARF and the remainder of this chapter is the previously existing text in the file. It is hence not verified and probably incorrect. This was working in the past and will probably be organized in some similar way in a future. 
```

File: documentation/smg/11_ISIS_release.md [hunk 15]
```diff
@@ -32,48 +48,44 @@ Code base for bugfix purposes is linked directly to the git repository for suppo
...
 
-#### Notes
+## Deploy Process
 
-The script which automates these operations and performs internal releases as one-click operation will be developed in the nearest future.
+As soon as release branch is tagged and created in the repository, manual build/release process is preformed in the private area of iDaaaS machine to obtain release artefacts for the iDAAaS computational service. It is also possible to pull the release artefacts from Jenkins builds, though this process needs thorough testing. 
+The extraction of the release artefacts will be automated in a future, as [iDaaaS testing pipeline](https://github.com/pace-neutrons/Horace/issues/271) is implemented.
 
-The members of **mslice** group on ISISCOMPUTE have write access to release code areas. Alex Buts or FBU IT support team can currently add users to **mslice** group on ISISCOMPUTE. Normal users have read-only access to the code area, sufficient to use the code in their computations.
+The releaser then manually checks out the release branch into `Horace_git` areas above and copies the release artefacts into appropriate places of the code tree.
```
Comment (cmarooney-stfc) [inline documentation/smg/11_ISIS_release.md:60]: I am finding this unclear. The exact processes to be used by the releaser to do this manual checkout and copy needs to be specified in detail.

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]
The corrections to spelling etc are minor but the details of what the releaser needs to do are the important corrections

File: documentation/smg/11_ISIS_release.md (hunk 5 above)
Comment (abuts) [reply documentation/smg/11_ISIS_release.md:12]: I hondestly do not know what they do but `pathdef.m` is unchanged. Probably they wrap Matlab launch script into into some other script or set up some enviromental variables.

File: documentation/smg/11_ISIS_release.md (hunk 6 above)
Comment (abuts) [reply documentation/smg/11_ISIS_release.md:14]: should be e.g. indeed.

File: documentation/smg/11_ISIS_release.md (hunk 11 above)
Comment (abuts) [reply documentation/smg/11_ISIS_release.md]: hopefully clarified

File: documentation/smg/11_ISIS_release.md (hunk 15 above)
Comment (abuts) [reply documentation/smg/11_ISIS_release.md:73]: added more about the release artefacts though they are descrined in other document

**Review by cmarooney-stfc — APPROVED** [review]
approving docs changes to test idaaas release

**Pull request merged**